content/docs/concepts/remote_write_spec_2_0.md

```diff
@@ -0,0 +1,394 @@
+}
+
+// Metadata represents the metadata associated with the given series' samples.
+message Metadata {
+  enum MetricType {
+    METRIC_TYPE_UNSPECIFIED        = 0;
+    METRIC_TYPE_COUNTER        = 1;
+    METRIC_TYPE_GAUGE          = 2;
+    METRIC_TYPE_HISTOGRAM      = 3;
+    METRIC_TYPE_GAUGEHISTOGRAM = 4;
+    METRIC_TYPE_SUMMARY        = 5;
+    METRIC_TYPE_INFO           = 6;
+    METRIC_TYPE_STATESET       = 7;
+  }
+  MetricType type = 1;
+  // help_ref is a reference to the Request.symbols array representing help
+  // text for the metric.
+  uint32 help_ref = 3;
+  // unit_ref is a reference to the Request.symbols array representing unit
+  // for the metric.
+  uint32 unit_ref = 4;
+}
+
+// A native histogram, also known as a sparse histogram.
+message Histogram { ... }
+
+// A BucketSpan defines a number of consecutive buckets with their
+// offset.
+message BucketSpan { ... }
+```
+
+All timestamps MUST be int64 counted as milliseconds since the Unix epoch. Sample's values MUST be float64.
+
+For every `TimeSeries` message:
+
+* Label references MUST be provided.
+* At least one element in Samples or in Histograms MUST be provided. For series which (rarely) would mix float and histogram samples, a separate `TimeSeries` message MUST be used.
+<!---
+TODO(bwplotka): We have some inconsistency here. In gdoc we talked this should be SHOULD. But proto has it as MUST (?). What do we want at the end? What's wrong with MUST here?
+-->
+* Metadata MUST be provided.
+* Exemplars SHOULD be provided, if they exist.
+* Created timestamp SHOULD be provided for metrics that follow counter semantics (e.g. counters and histograms).
+
+The following subsections define some schema elements in details.
+
+#### Symbols
+
+The `io.prometheus.write.v2.Request` proto schema is designed to [intern all strings](https://en.wikipedia.org/wiki/String_interning) for the proven additional compression and memory efficiency gains on top of the standard compressions.
+
+Symbols table containing deduplicated strings used in series and exemplar labels, metadata strings MUST be provided. References MUST point to the existing index in the Symbols string array.
+
+#### Series Labels
+
+The complete set of labels MUST be sent with each Sample or Histogram sample. Additionally, the label set associated with samples:
+
+- SHOULD contain a `__name__` label.
+- MUST NOT contain repeated label names.
+- MUST have label names sorted in lexicographical order.
+- MUST NOT contain any empty label names or values.
+
+Sender MUST only send valid metric names, label names, and label values:
+
+<!---
+TODO(bwplotka): Add mention for UTF-8 support going forward, see https://docs.google.com/document/d/1PljkX3YLLT-4f7MqrLt7XCVPG3IsjRREzYrUzBxCPV0/edit?disco=AAAA4gSqQ7g
+-->
+- Metric names MUST adhere to the regex `[a-zA-Z_:]([a-zA-Z0-9_:])*`.
+- Label names MUST adhere to the regex `[a-zA-Z_]([a-zA-Z0-9_])*`.
+- Label values MAY be any sequence of UTF-8 characters .
```

Comment:
do we want to keep this section, or instead just link to some spec for prometheus metrics themselves?